### ISSUE TITLE
Feat: Factory discoveries within a range of blocks

### ISSUE THREAD

**Issue by quangkeu95** [darkforestry/amms-rs#99]
Is it possible that we can call `discover_factories` with block params so i could run the discoveries within a range of blocks? Fetching from block 0 could be time consuming and inefficient.

**Comment by 0xOsiris** [comment]
Yeahh it's possible, but in the current architecture you'd have to change the code. Right now its defaulting with the `from_block` 0 and the `current_block` as the current chain head. If you want to discover within a specific range I think you should be able to just change out these block numbers to the range you want.
```rs

let mut from_block = 0;
let current_block = middleware
        .get_block_number()
        .await
        .map_err(AMMError::MiddlewareError)?
        .as_u64();

```

https://github.com/darkforestry/amms-rs/blob/00b267252db0237de82754882df09553d7684296/src/discovery/factory.rs#L50

**Comment by quangkeu95** [comment]
Yeah i've updated this feature in my branch. Should i create a PR?

**Comment by 0xKitsune** [comment]
Yes, thanks for opening an issue, this would be a good feature to add since there aren't any factories to discover for ~6 mm blocks. In your PR can you also update the default `from_block` to be a reasonable value as well?

**Comment by quangkeu95** [comment]
I've created a PR #102. I choose the `UniswapV2Factory` creation block as the default `from_block`, since I didn't know that any AMM has the same factory interface as UniswapV2Factory was created before. Please correct me if i'm wrong. Let me know your feedbacks as well. Thanks!

**Comment by 0xKitsune** [comment]
Cleaning up old issues, closing for now. This is addressed in the upcoming `0.7.0` release. Any factories that discover pools from logs[ will start from the factory creation block](https://github.com/darkforestry/amms-rs/blob/0.7.0/src/amms/uniswap_v3/mod.rs#L766) instead of block 0. 

#235 for reference
